ddsketch/mapping/logarithmic_mapping.go

```diff
@@ -45,36 +34,55 @@ func (m *LogarithmicMapping) Equals(other IndexMapping) bool {
 	if !ok {
 		return false
 	}
-	return (m.relativeAccuracy == o.relativeAccuracy && m.multiplier == o.multiplier)
+	return (withinTolerance(m.relativeAccuracy, o.relativeAccuracy, 1e-12) && withinTolerance(m.multiplier, o.multiplier, 1e-12))
 }
 
-func (m *LogarithmicMapping) Index(value float64) int {
+func (m *LogarithmicMapping) Index(value float64) int32 {
 	index := math.Log(value) * m.multiplier
 	if index >= 0 {
-		return int(index)
+		return int32(index)
 	} else {
-		return int(index) - 1 // faster than Math.Floor
+		return int32(index) - 1 // faster than Math.Floor
 	}
 }
 
-func (m *LogarithmicMapping) Value(index int) float64 {
+func (m *LogarithmicMapping) Value(index int32) float64 {
 	return math.Exp((float64(index) / m.multiplier)) * (1 + m.relativeAccuracy)
 }
 
 func (m *LogarithmicMapping) MinIndexableValue() float64 {
 	return math.Max(
-		math.Pow(math.E, minInt/m.multiplier+1), // so that index >= minInt
-		math.SmallestNonzeroFloat64*(1+m.relativeAccuracy)/(1-m.relativeAccuracy),
+		math.Exp(math.MinInt32/m.multiplier+1), // so that index >= MinInt32
+		minNormalFloat64*(1+m.relativeAccuracy)/(1-m.relativeAccuracy),
 	)
 }
 
 func (m *LogarithmicMapping) MaxIndexableValue() float64 {
 	return math.Min(
-		math.Pow(math.E, maxInt/m.multiplier-1), // so that index <= maxInt
+		math.Exp(math.MaxInt32/m.multiplier-1), // so that index <= MaxInt32
```

Comment:
This actually overflows `math.Exp` to `+Inf`, not sure what to set this to.

Reply:
Actually what overflows is `m.Value(m.Index(m.MaxIndexableValue()))`. The `math.Exp` function in `Value()` overflows to `+Inf`, similar for the underflow of `m.Value(m.Index(m.MinIndexableValue()))` in `LinearlyInterpolatedMapping`. What's confusing is that the code states that the `Exp` function should overflow over `7.097827...e+02` which should work with `math.MinInt32`, but it starts overflowing at `709.5`.

Reply:
Empirically determined the value at which `math.Exp` overflows and set the `MaxIndexableValue()` to prevent overflow.

Reply:
Ah sorry, I initially misunderstood. This makes sense to me now.